layerindex/tools/import_otherdistro.py

```diff
@@ -0,0 +1,789 @@
+                        if lastch == '%':
+                            if inshell == 0:
+                                shellexpr = ''
+                                outstr = outstr[:-1]
+                            inshell += 1
+                    if inmacro == 0 and inshell == 0:
+                        if ch == '%':
+                            # Handle unbracketed expressions (in which case we eat the rest of the expression)
+                            if expr[i+1] not in ['{', '%']:
+                                macrokey = expr[i+1:].split()[0]
+                                if macrokey in globaldefs or macrokey in defines or macrokey in values:
+                                    expanded = expand(values.get(macrokey, '') or defines.get(macrokey, '') or globaldefs.get(macrokey, ''))
+                                    if expanded:
+                                        outstr += expanded
+                                        break
+                        if ch == '%' and lastch == '%':
+                            # %% is a literal %, so skip this one (and don't allow this to happen again if the next char is a %)
+                            lastch = ''
+                            continue
+                        outstr += ch
+                    lastch = ch
+                return outstr
+
+            def eval_cond(cond, condtype):
+                negate = False
+                if condtype == '%if':
+                    if cond.startswith('(') and cond.endswith(')'):
+                        cond = cond[1:-1].strip()
+                    if cond.startswith('!'):
+                        cond = cond[1:].lstrip()
+                        negate = True
+                    res = False
+                    try:
+                        if int(cond):
+                            res = True
+                    except ValueError:
+                        pass
+                elif condtype in ['%ifos', '%ifnos']:
+                    if condtype == '%ifnos':
+                        negate = True
+                    # Assume linux
+                    res = ('linux' in cond.split())
+                elif condtype in ['%ifarch', '%ifnarch']:
+                    if condtype == '%ifnarch':
+                        negate = True
+                    res = ('x86_64' in cond.split())
+                else:
+                    raise Exception('Unhandled conditional type "%s"' % condtype)
+                if negate:
+                    return not res
+                else:
+                    return res
+
+            applypatches = {}
+            autopatch = None
+            conds = []
+            reading = True
+            inprep = False
+            applyextra = []
+            configopts = ''
+            inconf = False
+            pastpackage = False
+            for line in f:
+                if inconf:
+                    line = line.rstrip()
+                    configopts += line.rstrip('\\')
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are concatenating a string inside a loop. Concatenating immutable sequences will always results in a new object. This means that building up a sequence by repeated concatenation will have a quadratic runtime cost in the total sequence length. For efficiency and to get a linear runtime cost, consider appending values to a list and concatenating that list using [str.join()](https://docs.python.org/3/library/stdtypes.html#str.join) or [bytes.join()](https://docs.python.org/3/library/stdtypes.html#bytes.join) method.

[Learn more](https://docs.python.org/3/faq/programming.html#what-is-the-most-efficient-way-to-concatenate-many-strings-together)